plugins/azure/containerapps/externalNetworkAccess.js

```diff
@@ -0,0 +1,56 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'External Network Access Disabled',
+    category: 'Container Apps',
+    domain: 'Containers',
+    severity: 'Medium',
+    description: 'Ensure that Container Apps have external network access disabled.',
+    more_info: 'Disabling external network access for Container Apps ensures that inbound communication is restricted to callers within the apps environment, enhancing security by minimizing exposure to external threats. This helps safeguard sensitive data and prevents unauthorized access to the app resources.',
+    recommended_action: 'Update the Container Apps and disable external network access by enforcing internal-only ingress.',
+    link: 'https://learn.microsoft.com/en-us/azure/container-apps/ingress-how-to?pivots=azure-cli#ingress-settings',
+    apis: ['containerApps:list'],
+    realtime_triggers: ['microsoftapp:containerapps:write', 'microsoftapp:containerapps:delete'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var locations = helpers.locations(settings.govcloud);
+
+        async.each(locations.containerApps, function(location, rcb) {
+
+            var containerApps = helpers.addSource(cache, source,
+                ['containerApps', 'list', location]);
+
+            if (!containerApps) return rcb();
+
+            if (containerApps.err || !containerApps.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query for Container apps: ' + helpers.addError(containerApps), location);
+                return rcb();
+            }
+
+            if (!containerApps.data.length) {
+                helpers.addResult(results, 0, 'No existing Container apps found', location);
+                return rcb();
+            }
+
+            for (let container of containerApps.data) {
+                if (container.configuration.ingress && container.configuration.ingress.external) {
```

Comment:
add check for configuration i.e container.configuration && container.configuration.ingress